src/containers/Tenant/Schema/SchemaViewer/SchemaViewer.tsx

```diff
@@ -42,27 +40,28 @@ export const SchemaViewer = ({type, path, tenantName, extended = false}: SchemaV
     // Refresh table only in Diagnostics
     const pollingInterval = extended ? autoRefreshInterval : undefined;
 
-    const {currentData: schemaData, isFetching} = overviewApi.useGetOverviewQuery(
-        {path, database: tenantName},
-        {pollingInterval},
-    );
-
-    // useGetOverviewQuery is called in Tenant, so isLoading will be always false in this component
-    // to show loader properly, we need to check isFetching and currentData
-    const loading = isFetching && schemaData === undefined;
-
-    const viewSchemaRequestParams = isViewType(type) ? {path, database: tenantName} : skipToken;
-
-    const {currentData: viewColumnsData, isLoading: isViewSchemaLoading} =
-        viewSchemaApi.useGetViewSchemaQuery(viewSchemaRequestParams, {pollingInterval});
+    const {currentData: tableSchemaData, isFetching: isTableSchemaFetching} =
+        overviewApi.useGetOverviewQuery(
+            {path, database: tenantName},
+            {pollingInterval, skip: isViewType(type)},
+        );
+    const {currentData: viewColumnsData, isFetching: isViewSchemaFetching} =
+        viewSchemaApi.useGetViewSchemaQuery(
+            {path, database: tenantName},
+            {pollingInterval, skip: !isViewType(type)},
+        );
+
+    const loading =
+        (isViewSchemaFetching && viewColumnsData === undefined) ||
+        (isTableSchemaFetching && tableSchemaData === undefined);
```

Comment:
`isLoading` isn't set to `true` when `path` changes, so we don't see loader on schema object change